### ISSUE TITLE
[CoE Starter Kit - QUESTION] BYODL Sandbox environments

### ISSUE THREAD

**Issue by shoffma1** [microsoft/coe-starter-kit#6552]
### Does this question already exist in our backlog?

- [X] I have checked and confirm this is a new question.

### What is your question?

Hi,  Are Sanbox type environments expected to be in the Data Export to Azure Data Lake?

Seems they have been removed, and now we have at least one environment where several flows, but not all, have been marked as deleted.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.13

### What app or flow are you having the issue with?

Unknown

### What method are you using to get inventory and telemetry?

Data Export

**Comment by Jenefer-Monroe** [comment]
I do see Sandbox environments coming through in our BYODL test environment. 
If you go to the storage account > Containers > powerplatform > powerapps > environments > environments.json > Edit
Once there search for `"environmentType":"Sandbox"`
You can see if you have any in the file.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/c0a40cb0-1aad-4fb9-9dcb-0b11163945db)

**Comment by shoffma1** [comment]
We have not enabled governance and compliance yet, so not all of the flows are enabled.  But there are no solution layers at all.
Sorry for the tiny screenshot, trying to get all of the "Passed" items on one page ...
<img width="354" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/54184520/c1b9259a-ecbc-491c-be27-dc704b3512f4">

**Comment by Jenefer-Monroe** [comment]
Thank you, I just wanted to make sure all the ones in the "Inventory" solution area are on. Looks like they are so thats great.

Pick one of the Sandbox envts you expect to see.

Go to the latest run of this flow **_CoE BYODL - When Environment dataflow refresh is complete_**
Do you see the target Sandbox environment in the output of any of these calls?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/78e6397e-ce7d-454e-abfd-be86a5d8afd9)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/ab402418-eb67-4d2a-84a0-7c502a1de9c5)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/31aee332-58ec-4aa3-bf21-5d0d158881a0)

**Comment by shoffma1** [comment]
Yes, it is in the last one, only.

**Comment by Jenefer-Monroe** [comment]
ok so it is being found. Can you clarify why you think the envts arent coming through?

**Comment by shoffma1** [comment]
Sorry, it was just grasping at straws, looking at the org id rather than the environment id, in the data lake Flows folder.

We still have Flows from at least one environment missing, due to them marked as deleted, but they are still in the environment.  Only noticed because of exporting via REST API.

**Comment by Jenefer-Monroe** [comment]
Ok great lets start with a flow then instead that is incorrectly marked deleted. Note its envv GUID

Back in that storage account if you go to powerplatform > powerautomate > flows, find the correct file by looking for the envt guid, open it to view like you did above. Do you see the flow in the datalake file?

**Comment by shoffma1** [comment]
Yes, there are 44 flows listed in the datalake file, but only eight in the default "Active Flows" view when filtered on that environment. I'd send screenshots, but this is a fairly proprietary system.
Just removing the filter "Flow is Deleted" shows all 44 items in the Active Flows list in the admin view app.

**Comment by Jenefer-Monroe** [comment]
No need for the screenshot for this. just trying to narrow to where the issue is. 

Pick one of the flows thats incorrectly marked as deleted. 
Go to the last run of **_CoE BYODL - When Flow dataflow refresh is complete_**
And see if you can find your target flow in any of these spots
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/e54119bf-24c6-4d58-996d-7c7f953b4978)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/bb1386c4-18c9-4b42-9d1f-3b96201ec600)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/e60061be-66e1-4b0c-be2b-f05939dae36f)

**Comment by shoffma1** [comment]
Yes, lots of items in the "Update Missing Flows" -> "List flows that need updating", but also an error: 
Action failed: No dependent actions succeeded.

Also on "Get flow as admin" failed with error "Forbidden"

**Comment by Jenefer-Monroe** [comment]
Is the Power Platform Administrator Role permanently assigned? Or via PIM?

**Comment by Jenefer-Monroe** [comment]
Yes I mean the role as seen in Azure. Are the flows connections correctly attributed to this desired user with the role in Azure? And are they all showing green?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/a1ec4cfa-3fa6-4be1-a29c-59a93faaecd6)

**Comment by Jenefer-Monroe** [comment]
And they are showing for the identity running the flows (the identity that has the Power Platform Admin Role permanently assigned in Azure)

**Comment by shoffma1** [comment]
Yes, they are.

**Comment by Jenefer-Monroe** [comment]
Can you please show me an example here where the flow still exists but fails at this casll and goes down the "mark deleted" path?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/a7813b01-50c5-46c0-9088-2c08f227f5b6)

**Comment by shoffma1** [comment]
Not sure if this is the part you're asking for ... I'm not seeing the connectionReferences or other sections.

<img width="462" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/54184520/4817b8b6-d3e1-499b-935e-5f62b63b254b">

**Comment by shoffma1** [comment]
I think the issue here may be that our service account has not been provisioned for some of our environments.  Please correct me if I'm wrong, but even with the "Power Platform Administrator" role, the person/account still needs to have a role assigned to each dataverse environment?

**Comment by Jenefer-Monroe** [comment]
Thank you for debugging! 
Yes the Power Platform Admin Role is supposed to grant you system administrator access to any environment in the tenant when you make a request. 

Note that this requires that you 
1. Have the role permanently assigned (not via PIM)
2. Assigned directly to the individual not RBAC (something we didnt know about till yesterday so its not yet in our documentation, see https://github.com/microsoft/coe-starter-kit/issues/6592)

Now there was a recent product change here in which the product used to put you permanently in the Sys Admin role and now it does it dynamically so you may have hit that. 

Can you check for the two requirements above?

**Comment by shoffma1** [comment]
Fairly certain this is a direct assignment, but is part of being in a group ... 
<img width="834" alt="image" src="https://github.com/microsoft/coe-starter-kit/assets/54184520/e9849d81-af60-4a51-94bd-bfb7812963d1">

**Comment by Jenefer-Monroe** [comment]
Are you able to add your user directly instead of through the group? We've seen a handful of people hitting access issues and think it has to do with this.

**Comment by shoffma1** [comment]
We can certainly try a complete direct assignment.

However, I want to mention that I had manually, in a couple environments, added the "Service Reader" role to that account.  Now this morning, our service account does not have any roles in those environments, whereas it should at least have "System Administrator", yes?

We've seen something similar with regular users before in sandbox environments, but I believed that was due to an environment or app refresh.

I'm hoping, there should not be any reason, either through the CoE processes, or dataverse itself, that an environment would just erase security roles from an account?

**Comment by Jenefer-Monroe** [comment]
I"m afraid I dont know the answer to that. The kit does not remove people from SRs. You may want to contact product support for that.

**Comment by shoffma1** [comment]
Just FYI, one of two changes seems to have resolved the issue.
We got our service account the admin role directly assigned.  When that happened, I tried to resubmit the last failed run, but seemed to have the same results.
Looking at licenses for that account, I removed "Power Apps and Logic Flows".  
Today's flow run had just one environment access error, and that seems to be an environment that was deleted from our tenant.
